jax/lax/lax.py

```diff
@@ -1379,7 +1380,7 @@ def full(shape: Shape, fill_value: Array, dtype: Optional[DType] = None) -> Arra
   return broadcast(fill_value, shape)
 
 def _device_put_raw(x):
-  if isinstance(x, xla.DeviceValue):
+  if isinstance(x, xla.DeviceValue) or numpy_eval.is_tracing:
```

Comment:
I'm not sure yet what the right fix to this is, but one thing we could do is just check if `core.thread_local_state.trace_state.trace_stack.stack[0]` is a `NumpyEvalTrace`. We could use that for now and leave a TODO(mattjj) to figure out something better.

Reply:
Done!